main/core/injector.ts

```diff
@@ -13,50 +13,77 @@ import {
     IInjectionToken,
     IInjector,
     IMetrics,
-    IScopeConfiguration,
+    IParameterInjectionToken,
     ITokenCache,
     Newable,
 } from '../contracts/contracts';
-import { createInstance } from '../internal/construction';
 import { InstanceCache } from './cache';
 import { Configuration } from './configuration';
 import { AutoFactory } from './factory';
-import { getGlobal, globalState } from './globals';
+import { getGlobal } from './globals';
+import { isFactoryParameterToken, isLazyParameterToken } from './guards';
 import { LazyInstance } from './lazy';
 import { getConstructorTypes } from './metadata.functions';
 import { Metrics } from './metrics';
 import { InjectionTokensCache } from './tokens';
 
-const globalReference = getGlobal();
-
 /**
  * Injector type used for registering types for injection
  */
 export class Injector implements IInjector {
+    /**
+     * Creates a new map from our internal child maps
+     */
+    public get children(): Map<InjectorIdentifier, IInjector> {
+        return new Map(this._children);
+    }
+
+    /**
+     * @internal Create a new injector instance for this type (Could change in the future for different versions)
+     * @description When multiple versions of the injector are in play, this function ensures we can
+     * construct it without doing the work externally. (no introspection required).
+     */
+    public static create(
+        parent?: IInjector,
+        name?: string,
+        configuration: Configuration = new Configuration(),
+    ): Injector {
+        return new Injector(
```

Comment:
I can't see how the children array of the parent gets updated here.

Reply:
Creating a scope adds a child. Destroying injector kills children. Covered by the tests. 

Maybe clone the pull request and run it. Should be easier to see